staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_filter.go

```diff
@@ -46,12 +45,17 @@ import (
 
 // Interface defines how the API Priority and Fairness filter interacts with the underlying system.
 type Interface interface {
-	// Wait decides what to do about the request with the given digest
-	// and, if appropriate, enqueues that request and waits for it to be
-	// dequeued before returning.  If `execute == false` then the request
-	// is being rejected.  If `execute == true` then the caller should
-	// handle the request and then call `afterExecute()`.
-	Wait(ctx context.Context, requestDigest fcfc.RequestDigest) (execute bool, afterExecute func())
+	// Handle takes care of queuing and dispatching a request
+	// characterized by the given digest.  The given `noteFn` will be
+	// invoked with the results of request classification.  If Handle
+	// decides that the request should be executed then `execute()`
+	// will be invoked once to execute the request; otherwise
+	// `execute()` will not be invoked.
+	Handle(ctx context.Context,
+		requestDigest RequestDigest,
+		noteFn func(fs *fctypesv1a1.FlowSchema, pl *fctypesv1a1.PriorityLevelConfiguration),
```

Comment:
If I understand it correctly, noteFn provides a way of notification. What bothers me is in what situation that the invoker might cares about it, maybe for metrics or logging? Take the priority-and-fairness server handler for example, in its entire request process, it doesn't need to know the existence of fs and plc. What's the purpose of expose noteFn?

Reply:
We plan for a later PR to enhance the request handler to add the FlowSchema and PriorityLevelConfiguration UIDs to the response headers.